addons/stock/models/stock_inventory.py

```diff
@@ -211,28 +160,77 @@ def action_cancel_draft(self):
         })
 
     def action_start(self):
-        for inventory in self.filtered(lambda x: x.state not in ('done','cancel')):
-            vals = {'state': 'confirm', 'date': fields.Datetime.now()}
-            if (inventory.filter != 'partial') and not inventory.line_ids:
-                vals.update({'line_ids': [(0, 0, line_values) for line_values in inventory._get_inventory_lines_values()]})
-            inventory.write(vals)
-        return True
+        for inventory in self:
+            # Confirms the inventory adjustment and generates its lines if inventory
+            # state is draft
+            if inventory.state == 'draft':
+                vals = {
+                    'state': 'confirm',
+                    'date': fields.Datetime.now()
+                }
+                if not inventory.line_ids and not inventory.start_empty:
+                    vals['line_ids'] = [(0, 0, line_values) for line_values in inventory._get_inventory_lines_values()]
+                inventory.write(vals)
+            return inventory.action_open_inventory()
```

Comment:
this method looks like it is working on multiple inventories (for inventory in self) but at the end returns an action
as this method is called by a button in a form view,i think we can expect an ensure_one